<!-- This launch file executes flocking algorithm for 2 robots, random walk type - ballistic motion -->

<launch>
    <arg name="x" value="0"/>
    <arg name="y" value="5"/>
    <arg name="r_num" value="2"/>

    <!--ENVIRONMENT-->
    <param name="rbts_num" value="$(arg r_num)" />
    <node pkg='turtlebot3_motion' type="environment_flocking.py" name="env" output="screen" />

    <!--ROBOT 0-->
    <group ns="r0">
        <arg name="num" value="0"/>
        <param name="num" value="$(arg num)"/>
        <param name="n_num" value="1" />
        <param name="des_pos_x" value="$(arg x)"/>
        <param name="des_pos_y" value="$(arg y)"/>
        <param name="rbts_num" value="$(arg r_num)" />
        <node pkg="turtlebot3_motion" type="flocking.py" name="r$(arg num)_flocking" output="screen" />
        <node pkg="turtlebot3_motion" type="flocking_go_to_point.py" name="r$(arg num)_flocking_go_to_point" output="screen" />
        <node pkg="turtlebot3_motion" type="follow_the_wall.py" name="r$(arg num)_follow_the_wall" output="screen" />
        <node pkg="turtlebot3_motion" type="ballistic_motion.py" name="r$(arg num)_random_walk" output="screen" />

    </group>
    
    <!--ROBOT 1-->
    <group ns="r1">
        <arg name="num" value="1"/>
        <param name="num" value="$(arg num)" />
        <param name="n_num" value="0"/>
        <param name="des_pos_x" value="$(arg x)"/>
        <param name="des_pos_y" value="$(arg y)"/>
        <param name="rbts_num" value="$(arg r_num)" />
        <node pkg="turtlebot3_motion" type="flocking.py" name="r$(arg num)_flocking" output="screen" />
        <node pkg="turtlebot3_motion" type="flocking_go_to_point.py" name="r$(arg num)_flocking_go_to_point" output="screen" />
        <node pkg="turtlebot3_motion" type="follow_the_wall.py" name="r$(arg num)_follow_the_wall" output="screen" />
        <node pkg="turtlebot3_motion" type="ballistic_motion.py" name="r$(arg num)_random_walk" output="screen" />
    </group>

</launch>